### PR TITLE
refactor: valid_date does not work in PHP 8.2

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
See #6170
See https://github.com/codeigniter4/CodeIgniter4/pull/6172#issuecomment-1275496541

```console
bash-3.2$ ./phpunit tests/system/Validation/FormatRulesTest.php
PHPUnit 9.5.25 #StandWithUkraine

Runtime:       PHP 8.2.0-dev
Configuration: /Users/kenji/work/codeigniter/official/CodeIgniter4/phpunit.xml

...............................................................  63 / 193 ( 32%)
............................................................... 126 / 193 ( 65%)
............................................................... 189 / 193 ( 97%)
....                                                            193 / 193 (100%)

Time: 00:00.325, Memory: 14.00 MB

OK (193 tests, 193 assertions)
```

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Validation/FormatRules.php [hunk 1]
```diff
@@ -344,6 +344,11 @@ public function valid_date(?string $str = null, ?string $format = null): bool
         $date   = DateTime::createFromFormat($format, $str);
         $errors = DateTime::getLastErrors();
 
+        // PHP 8.2 or later.
+        if ($date !== false && $errors === false) {
```
Comment (samsonasik) [inline system/Validation/FormatRules.php:348]: `$date === false` or `! $date instanceof DateTime` should can be returned early

**Comment by paulbalandan** [comment]
Is there a case where `$date` is a `DateTime` instance yet `$errors` contains actual errors? If no, then can we remove the `getLastErrors` check?

File: system/Validation/FormatRules.php (hunk 1 above)
Comment (kenjis) [reply system/Validation/FormatRules.php]: Done!

**Comment by kenjis** [comment]
> Is there a case where $date is a DateTime instance yet $errors contains actual errors?

Yes.
```php
            [
                '1991-02-29',
                'Y-m-d',
                false,
            ],
            [
                'Thu, 31 Jun 2013 13:31:00',
                'D, d M Y H:i:s',
                false,
            ],
            [
                '2, 17, 2001',
                'j, n, Y',
                false,
            ],
            [
                '09-42-25, 12-00-17',
                'h-i-s, j-m-y',
                false,
            ],
```

**Review by paulbalandan — APPROVED** [review]

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @kenjis

**Pull request merged**